src/mapping/htseq_count/script.py

```diff
@@ -0,0 +1,114 @@
+import tempfile
+import subprocess
+from pathlib import Path
+import tarfile
+import gzip
+import shutil
+import yaml
+
+## VIASH START
+par = {
+  'input': ['resources_test/cellranger_tiny_fastq/bam/possorted_genome_bam.bam'],
+  'reference': 'resources_test/cellranger_tiny_fastq/cellranger_tiny_ref/genes/genes.gtf.gz',
+  'output': 'test_output'
+}
+meta = {
+  'cpus': 2,
+  'temp_dir': '/tmp',
+  'config': 'src/mapping/htseq/config.vsh.yaml'
+}
+## VIASH END
+
+########################
+### Helper functions ###
+########################
+
+# helper function for cheching whether something is a gzip
+def is_gz_file(path: Path) -> bool:
+  with open(path, 'rb') as file:
+    return file.read(2) == b'\x1f\x8b'
+
+# if {par_value} is a Path, extract it to a temp_dir_path and return the resulting path
+def extract_if_need_be(par_value: Path, temp_dir_path: Path) -> Path:
+  if par_value.is_file() and tarfile.is_tarfile(par_value):
```

Comment:
Could you try to use viashpy for this (only the tarfile)?

Reply:
I will if you can also provide the `extract_if_need_be` functionality :D 

```python
# if {par_value} is a Path, extract it to a temp_dir_path and return the resulting path
def extract_if_need_be(par_value: Path, temp_dir_path: Path) -> Path:
  if par_value.is_file() and tarfile.is_tarfile(par_value):
    print(f'  Tar detected; extracting {par_value}', flush=True)
    return extract_tar(par_value, temp_dir_path)

  elif par_value.is_file() and is_gz_file(par_value):
    # Remove extension (if it exists)
    extaction_file_name = Path(par_value.stem)
    unpacked_path = temp_dir_path / extaction_file_name
    print(f'  Gzip detected; extracting {par_value}', flush=True)
    
    with gzip.open(par_value, 'rb') as f_in:
      with open(unpacked_path, 'wb') as f_out:
        shutil.copyfileobj(f_in, f_out)
    return unpacked_path

  else:
    return par_value
```

Ty! :bow: